persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/EditFilenameService.java

```diff
@@ -67,17 +69,21 @@ public void editLabel(AgentPrincipals agent, PID pid, String label) {
 
         try (Timer.Context context = timer.time()) {
             aclService.assertHasAccess(
-                    "User does not have permissions to edit labels",
+                    "User does not have permissions to edit filenames",
                     pid, agent.getPrincipals(), Permission.editDescription);
 
             RepositoryObject obj = repoObjLoader.getRepositoryObject(pid);
 
+            if (!(obj instanceof FileObject)) {
+                throw new Exception("Failed to edit filename for " + obj.getPid());
+            }
+
             String oldLabel = getOldLabel(obj);
 
             repoObjFactory.createExclusiveRelationship(obj, DcElements.title, label);
```

Comment:
Its occurring to me now that we should change the filename instead of the label. Could you change this to setting the ebucore:filename? We retain the original filename in the PREMIS, so that should be okay.